aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionEndpoint.java

```diff
@@ -195,6 +200,10 @@ private void processJsonMessage(WebSocket webSocket, String message) throws ApiE
                     );
                     connectionAcknowledgement.countDown();
                     break;
+                case CONNECTION_ERROR:
+                    connectionFailure = message;
+                    connectionAcknowledgement.countDown();
```

Comment:
This variable was named `connectionAcknowledgement` since it was in response to a `connection_ack` message type. If we use it for two distinct (though closely related purposes), perhaps it should be renamed as `connectionResponsesPending`, or something.

Reply:
Makes sense